src/sage/rings/polynomial/polynomial_element.pyx

```diff
@@ -5523,6 +5523,18 @@ cdef class Polynomial(CommutativePolynomial):
         if tgt.ngens() > 1 and tgt._has_singular:
             g = flatten(self).gcd(flatten(other))
             return flatten.section()(g)
+        try:
+            P = self.parent()
+            g = self._pari_with_name().gcd(other._pari_with_name())
+            g = P(g)
+            c = g.leading_coefficient()
+            if c != P(0):
+                g /= c
+            return g
+        except (PariError, ZeroDivisionError):
+            raise ValueError("Failed to calculate GCD on polynomials over composite ring.")
```

Comment:
```suggestion
            raise ValueError("failed to calculate GCD on polynomials over composite ring")
```

(this is the convention although there are still 1521 occurrences in the codebase that uppercase the first character)